os_stub/spdm_device_secret_lib_sample/lib.c

```diff
@@ -28,6 +28,789 @@
+    0xa5, 0x62, 0xfd, 0xd4, 0x2d, 0x21, 0x04, 0x5d, 0xa3, 0xc1, 0x7a, 0xeb, 0x3f, 0x77, 0xac,
+    0xc7, 0x18, 0x35, 0x14, 0xd0, 0x78, 0xbd, 0xdb, 0xf0, 0x70, 0x5b, 0xbe, 0xd7, 0x05, 0x27,
+    0x48, 0x3d, 0xd9, 0xd3, 0x42, 0x6b, 0xd1, 0x33, 0xb6, 0x08, 0x79, 0xd2, 0x07, 0xaf, 0xf6,
+    0xdc, 0x77, 0xf7, 0x96, 0x04, 0xed, 0xf2, 0x44, 0x45, 0xae, 0x07, 0xb7, 0x20, 0x2b, 0x11,
+    0x29, 0x43, 0xe3, 0x3c, 0xc1, 0x59, 0x19, 0xb1, 0x62, 0xff, 0xb4, 0x3e, 0xfa, 0x26, 0xb6,
+    0xf4, 0xca, 0xe3, 0xbd, 0x8f, 0xb5, 0x35, 0x6c, 0xec, 0xd1, 0xd9, 0x25, 0xd6, 0x63, 0xaf,
+    0xbb, 0x17, 0x9e, 0x53, 0x31, 0x23, 0x52, 0x47, 0x1c, 0x9c, 0x68, 0x56, 0x9e, 0x4c, 0xb3,
+    0x90, 0xb9, 0x01, 0x7c, 0x8d, 0xf5, 0xc5, 0x32, 0xff, 0xef, 0x6e, 0x78, 0x04, 0x48, 0x81,
+    0xd9, 0x7d, 0x0f, 0x73, 0x67, 0x04, 0x49, 0x8a, 0xec, 0x3f, 0x0c, 0x6b, 0x56, 0x38, 0xa9,
+    0x6e, 0x09, 0x1c, 0x22, 0xd3, 0x65, 0x59, 0xc7, 0xdd, 0xfd, 0xa6, 0x3a, 0xad, 0xbb, 0xb2,
+    0x13, 0x12, 0x9b, 0xb4, 0x7e, 0xf4, 0xf5, 0x59, 0x0d, 0x96, 0x7e, 0x7f, 0x0f, 0x37, 0x42,
+    0x6a, 0x7b, 0x73, 0x41, 0xb6, 0x08, 0xbe, 0x24, 0x54, 0x02, 0xac, 0x18, 0x15, 0xd3, 0x96,
+    0x98, 0xd3, 0x1e, 0x64, 0x3c, 0xd0, 0xbc, 0x4e, 0xfd, 0xea, 0xe9, 0xb4, 0x14, 0xc3, 0x58,
+    0x11, 0x6f, 0xd6, 0x35, 0xfe, 0x58, 0x06, 0x11, 0xc9, 0xfe, 0x9e, 0xd5, 0x43, 0x0e, 0x08,
+    0xb9, 0x7b, 0xc7, 0x40, 0x55, 0x44, 0xfe, 0x3c, 0x05, 0x58, 0x94, 0x66, 0xe1, 0x1a, 0x7b,
+    0x16, 0x67, 0x95, 0x48, 0x4b, 0x5c, 0xbc, 0xa8, 0xcc, 0xd8, 0x9f, 0xb5, 0xac, 0xd2, 0x71,
+    0x70, 0x8a, 0x3b, 0x3c, 0x5f, 0x38, 0x4b, 0x4d, 0x9b, 0x83, 0x3f, 0x41, 0x95, 0x95, 0x92,
+    0xf2, 0x4f, 0x03, 0xd3, 0xba, 0xc6, 0xa4, 0xe9, 0xcf, 0x52, 0xf0, 0xee, 0xfd, 0xe5, 0xf5,
+    0x94, 0x93, 0x78, 0xa9, 0xe5, 0x81, 0x91, 0xe3, 0xda, 0x88, 0x1b, 0x8e, 0x73, 0xfe, 0x4a,
+    0x0c, 0xd5, 0x5e, 0xef, 0xcf, 0x65, 0x72, 0xd2, 0x73, 0x51, 0x0d, 0x37, 0x1c, 0x14, 0xf0,
+    0x1d, 0x6d, 0x25, 0xe9, 0x50, 0x47, 0xcc, 0xa7, 0xb2, 0x60, 0x89, 0x7c, 0xef, 0xd2, 0xb9,
+    0x5e, 0x34, 0xb8, 0x36, 0x63, 0xcd, 0xaf, 0xdf, 0x55, 0x0f, 0xd4, 0xfa, 0x34, 0x66, 0x60,
+    0xe8, 0x58, 0x0b, 0xa5, 0x4d, 0x00, 0xf0, 0xd4, 0x45, 0x8e, 0x49, 0xb3, 0x16, 0x56, 0x49,
+    0xd3, 0x26, 0x39, 0x65, 0x26, 0x76, 0xfc, 0x1f, 0xdb, 0x77, 0x29, 0x6f, 0xfe, 0x84, 0x71,
+    0x68, 0x49, 0xa1, 0x9e, 0x74, 0x69, 0x4e, 0x42, 0x24, 0x11, 0x47, 0x3c, 0x96, 0x2b, 0xc6,
+    0x0b, 0xf5, 0x1c, 0x5f, 0xfe, 0xc6, 0x7e, 0xab, 0xd6, 0x19, 0x60, 0x6b, 0x99, 0x9e, 0x66,
+    0x5f, 0xbc, 0xf6, 0xb5, 0xb5, 0x41, 0x01, 0x91, 0x6a, 0xc4, 0x3e, 0x03, 0x5a, 0x67, 0x6d,
+    0xf6, 0x34, 0x8a, 0xff, 0x87, 0xa9, 0x9e, 0x84, 0x6e, 0x2b, 0x45, 0xb0, 0x88, 0x96, 0xb5,
+    0x08, 0x01,
+};
+
+#endif
+
+bool libspdm_get_responder_private_key_from_raw_data(uint32_t base_asym_algo, void **context)
+{
+    bool result;
+    void *ec_context;
+    void *rsa_context;
+
+    switch (base_asym_algo) {
+    case SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_RSASSA_2048:
```

Comment:
please use LIBSPDM_RSA_SSA_SUPPORT, LIBSPDM_RSA_PSS_SUPPORT, LIBSPDM_ECDSA_SUPPORT to conditional include those features.

Reply:
This switch case does not have MACRO.
I think it will fail to build if set RSA or ECDSA to 0.

Please do the test to ensure it can pass build.